### PR TITLE
Testing Kirkwood

### PR DESCRIPTION
**Pull request by ckirkwood94**
@gooseandmegander

### CODE REVIEW

File: basic-javascript/project-1.spec.js [hunk 1]
```diff
@@ -1,3 +1,234 @@
...
 
 // start testing!
+describe('project-1.js', () => {
+  describe('mutiplyByTen', () => {
+    it('should multiply num by 10', () => {
+      const expected = 30;
+      const actual = helpers.multiplyByTen(3);
+      expect(actual).toEqual(expected);
+    });
+  });
```
Comment (gooseandmegander) [inline basic-javascript/project-1.spec.js:11]: What if a string is supplied? Or nothing?

File: basic-javascript/project-1.spec.js [hunk 2]
```diff
@@ -1,3 +1,234 @@
...
+
+  describe('subtractFive', () => {
+    it('should subtract 5 from num', () => {
+      const expected = 30;
+      const actual = helpers.subtractFive(35);
+      expect(actual).toEqual(expected);
+    });
+  });
```
Comment (gooseandmegander) [inline basic-javascript/project-1.spec.js:19]: Depending on the context, should the output be allowed to be negative?

File: basic-javascript/project-1.spec.js [hunk 3]
```diff
@@ -1,3 +1,234 @@
...
+      const expected = true;
+      const actual = helpers.areSameLength('str1', 'abcd');
+      expect(actual).toEqual(expected);
+    });
+    it('should return false if strings are not equal length', () => {
+      const expected = false;
+      const actual = helpers.areSameLength('not equal', 'longer string');
+      expect(actual).toEqual(expected);
+    });
+  });
```
Comment (gooseandmegander) [inline basic-javascript/project-1.spec.js:32]: What if a boolean is supplied?

File: basic-javascript/project-1.spec.js [hunk 4]
```diff
@@ -1,3 +1,234 @@
...
+
+  describe('divide', () => {
+    it('should divide first number by second number', () => {
+      const expected = 9;
+      const actual = helpers.divide(90, 10);
+      expect(actual).toEqual(expected);
+    });
+  });
```
Comment (gooseandmegander) [inline basic-javascript/project-1.spec.js:95]: What if the denominator is zero?

File: basic-javascript/project-1.spec.js [hunk 5]
```diff
@@ -1,3 +1,234 @@
...
+
+  describe('addExclamationPoint', () => {
+    it('should add exclamation point to string', () => {
+      const expected = 'Hello!';
+      const actual = helpers.addExclamationPoint('Hello');
+      expect(actual).toEqual(expected);
+    });
+  });
```
Comment (gooseandmegander) [inline basic-javascript/project-1.spec.js:185]: What if a number is supplied?

File: basic-javascript/project-2.spec.js [hunk 6]
```diff
@@ -1,3 +1,217 @@
...
 
 // whoops.. there is no test suite for this file. You'll simply just have to create one :/
+describe('project-2.js', () => {
+  describe('getBiggest', () => {
+    it('should return larger number', () => {
+      const expected = 30;
+      const actual = funcs.getBiggest(30, 20);
+      expect(actual).toEqual(expected);
+    });
+  });
```
Comment (gooseandmegander) [inline basic-javascript/project-2.spec.js:11]: What if a string is supplied?

**Review by gooseandmegander — COMMENTED** [review]
# Objectives
* Familiarity with writing unit tests for JavaScript code.

# MVP
- [x] project-1.spec.js tests passing
- [x] project-2.spec.js tests passing

# Stretch
- [ ] arrays.spec.js tests passing
- [ ] objects.spec.js tests passing

# Great
All the tests pass. There are "green" and "red" tests.

# Requests for Improvement
* Write more "red" tests for cases that shouldn't pass. How will different primitive values affect the output? How does the context of the test affect what the inputs _ought_ to be? Etc. *Why would you want to keep your tests even though what you're testing is passing those tests?*
* Check out other types of testing.
* Use something new from the docs you haven't used yet to generate tests.
* Think about edge cases.

# Rating: 2
